### ISSUE TITLE
ZBS runs into an error loop, when closing many files

### ISSUE THREAD

**Issue by jjvbsag** [pkulchenko/ZeroBraneStudio#968]
Ok, this is a strange one:
When I have many files open and I want to clean up my view, my habbit is to press and hold [Ctrl]+[W], until all windows are closed. Starting from an uncertain amount of open windows, ZBS gets mad and begins to create endless message boxes:
![image](https://user-images.githubusercontent.com/8567887/51388360-74481180-1b29-11e9-8e5d-a6d19645e6cd.png)
There is no chance to get rid of them. ZBS is faster to create new ones :imp: The only way is to somehow manage to logout. You cannot even create a terminal window to kill ZBS because the messageboxes always get the input focus.

If you will try yourself:

1. Create many files (with some amount of text) with this script

```lua
#!/usr/bin/lua
for i=1,100 do
	local fd=assert(io.open("script"..i..".lua","w"))
	for l=1,100 do
		fd:write("print(",i,",",l,") -- asddddddddddddddddddddddddddddddd\n")
	end
	fd:close()
end
```

2. Open all of them with
`zbstudio *.lua`

3. Try to close the windows in ZBS by press and hold [Ctrl]+[W]

If you get no problem, increase the loop value for `l`

**Comment by pkulchenko** [comment]
@jjvbsag, thank you for the report. I expect the following patch to fix the issue:

```diff
diff --git a/src/editor/editor.lua b/src/editor/editor.lua
index 0b97d734..2e4a8f0b 100644
--- a/src/editor/editor.lua
+++ b/src/editor/editor.lua
@@ -436,13 +436,6 @@ end
 
 local delayed = {}
 
-function IndicateIfNeeded()
-  local editor = ide:GetEditor()
-  -- do the current one first
-  if delayed[editor] then return editor:IndicateSymbols() end
-  for ed in pairs(delayed) do return ed:IndicateSymbols() end
-end
-
 -- find all instances of a symbol at pos
 -- return table with [0] as the definition position (if local)
 local function indicateFindInstances(editor, name, pos)
@@ -631,6 +624,13 @@ local function indicateSymbols(editor, lines)
   return needmore -- request more events if still need to work
 end
 
+function IndicateIfNeeded()
+  local editor = ide:GetEditor()
+  -- do the current one first
+  if delayed[editor] then return editor:IndicateSymbols() end
+  for ed in pairs(delayed) do return indicateSymbols(ed) end
+end
+
 -- ----------------------------------------------------------------------------
 -- Create an editor
 function CreateEditor(bare)
```

Let me know.

**Comment by pkulchenko** [comment]
You'll also need to upgrade to the current master, as this issue is already fixed in e67f82157.
